libs/wingsdk/src/platform/platform-manager.ts

```diff
@@ -1,129 +1,234 @@
+   * deployed into a single environment.
+   *
+   * @default - no custom root id
+   */
+  readonly customRootId?: string;
+}
+
+const BUILTIN_PLATFORMS = ["tf-aws", "tf-azure", "tf-gcp", "sim"];
+
+/** @internal */
+export const SECRETS_FILE_NAME = "secrets.json";
+
+/** @internal */
+export class PlatformManager implements IPlatformManager {
+  public static load(target?: string): IPlatformManager {
+    const extensions = [];
+    if (target) {
+      extensions.push(target);
+    }
+    extensions.push(...(process.env.WING_PLATFORMS?.split(";") ?? []));
 
-  private createPlatformInstances() {
-    this.platformInstances = [];
-    this.platformPaths.forEach((platformPath) => {
-      this.loadPlatformPath(platformPath);
+    return new PlatformManager({
+      extensions,
+      customRootId: process.env.WING_ROOT_ID,
     });
   }
 
-  // This method is called from preflight.cjs in order to return an App instance
-  // that can be synthesized
-  public createApp(appProps: AppProps): App {
-    this.createPlatformInstances();
-    let appCall = this.platformInstances[0].newApp;
+  // private readonly platformPaths: string[];
+  private readonly extensions: IPlatformProvider[] = [];
+  public readonly parameters: ParameterRegistrar;
+  public readonly provider: IPlatformProvider;
+  public readonly target: string;
 
-    if (!appCall) {
+  private readonly customRootId?: string;
+
+  private constructor(options: PlatformManagerOptions) {
+    this.customRootId = options.customRootId;
+
+    const specs = [...options.extensions, ...retrieveImplicitPlatforms()];
+    this.extensions = specs.map((path) => loadPlatform(path));
+    this.parameters = new ParameterRegistrar();
+    if (!this.extensions[0]) {
       throw new Error(
-        `No newApp method found on platform: ${this.platformPaths[0]} (Hint: The first platform provided must have a newApp method)`
+        `No platform found (Hint: Ensure you have installed a platform provider)`
+      );
+    }
+
+    this.provider = this.extensions[0];
+    if (!this.provider) {
+      throw new Error("At least one platform specifier must be provided");
+    }
+
+    this.target = this.provider.target;
+  }
+
+  public resolveType(fqn: string) {
+    return this.provider.resolveType(fqn);
+  }
+
+  public newAbstract(fqn: string, ...args: any[]) {
+    const type = this.resolveType(fqn);
+    if (!type) {
+      const typeName = fqn.replace(`${SDK_PACKAGE_NAME}.`, "");
+      const typeNameParts = typeName.split(".");
+      throw new NotImplementedError(
+        `Resource "${fqn}" is not yet implemented for "${this.provider.target}" target. Please refer to the roadmap https://github.com/orgs/winglang/projects/3/views/1?filterQuery=${typeName}`,
+        { resource: typeNameParts[typeNameParts.length - 1] }
       );
     }
 
-    let hooks = collectHooks(this.platformInstances);
+    return new type(...args);
+  }
+
+  public synth(rootConstruct: any): void {
```

Comment:
I think bringing `synth` into platform manager is overall a good idea, the only downside I can think of is it now removes the ability to customize the `synth` process on a per platform level. Which is maybe a good thing, as I think this will resolve issues like: https://github.com/winglang/wing/issues/6733

Plus it ensures the extension hooks for synthesis will be called in the correct order.